src/DynamoCoreWpf/ViewModels/PackageManager/PackageViewModel.cs

```diff
@@ -99,7 +113,25 @@ public string PackageViewContextMenuUninstallText
 
         public string PackageViewContextMenuUninstallTooltip
         {
-            get { return Model.BuiltInPackage ? Resources.PackageContextMenuUnloadPackageTooltip : Resources.PackageContextMenuDeletePackageTooltip; }
+            get
+            {
+                // Built in package
+                if (Model.BuiltInPackage)
+                {
+                    return Resources.PackageContextMenuUnloadPackageTooltip;
+                }
+
+                // Package with custom nodes that are in use
+                if (Model.LoadState.State != PackageLoadState.StateTypes.Unloaded
+                    && Model.LoadState.ScheduledState != PackageLoadState.ScheduledTypes.ScheduledForDeletion
+                    && !CanUninstall())
+                {
+                    return Resources.PackageContextMenuDeletePackageCustomNodesInUseTooltip;
```

Comment:
@sm6srw  By custom nodes ...do you mean non-built-in ? or custom node definitions ?

Reply:
Custom Node definitions

Reply:
Hmm..I do not see any specific checks for custom node defs...is that the only possibility?...what about packages with loaded assemblies...

Reply:
The check is in CanUnInstall(). It is a little strange because we say that a package can always be uninstalled if the package has a loaded assembly. But it can not be uninstalled if it has custom nodes that are in use. We missed that case. What I think is missing is to do the same thing for all packages (allow or not allow). But that is future work. For now we just try to clarify this a bit.